assets/js/ajax_search_select.js

```diff
@@ -0,0 +1,168 @@
+                            self.field.focus();
+                        }, 500);
+                    }
+                });
+            this.search = this.widget.find(this.settings.search_selector)
+                .on("show.bs.dropdown", function(event) {
+                    self.searchOptions();
+                });
+            this.element.parents("form").on("submit", function(event) {
+                self.finish();
+            });
+            this.element.removeAttr('required');
+        },
+
+        searchOptions: function(show_dropdown) {
+            if (show_dropdown && this.result.is(":visible") === false) {
+                this.search.find("button").dropdown("toggle");
+            }
+            var selector = "option";
+            const query = this.field.val().trim();
+            const self = this;
+            if (query.length > 0) {
+              $.ajax({
+                url: this.api_url,
+                data: {
+                  "search": query,
+                },
+              }).done(function(data) {
+                const numOfEntries = data.count;
+                self.result.empty();
+                if (numOfEntries > 0) {
+                  for (let i = 0; i < numOfEntries; i++) {
+                      const result_info = data.results[i];
+                      self.result.append(
+                        $('<li>').append(
+                          $('<a>').text(
+                            self.parameter_list.map(
+                                item => result_info[item]
+                            ).join(', ')
+                          )
+                        ).click(function() {
+                          self.addSelection(
+                            result_info['id'], self.parameter_list.map(
+                                item => result_info[item]
+                            ).join(', ')
+                          )
+                        })
+                      );
+                    }
+                }
+                else {
+                  self.result.append(
+                    $('<li>').append(
+                      $('<a>').text(_("No matches!"))
+                  ));
+            }
+            });
+          }
+        },
+
+        addSelection: function(value, name) {
+            if (this.selection.find('[data-value="' + value + '"]').size() === 0) {
+                var li = this.selection_li.clone();
+                li.find(".name").text(name);
+                li.find("button").attr("data-value", value).on('click', function(event) {
+                    $(this).parent("li").remove();
+                });
+                this.selection.append(li);
+            }
+        },
+
+        resetSelection: function(values) {
+            this.selection.empty();
+            var self = this;
+            $.each(values, function(index, value) {
+                var opt = self.element.find('option[value="' + value + '"]');
```

Comment:
Does this `resetSelection` method work in the new AJAX version? It appears this method is not often called.

Reply:
I don't think that it works right now, but I have to admit that I don't know in which scenarios it is called and why.
It is probably useless in this new code.

Reply:
You can see in the bottom that the jQuery extension may call resetSelection if you supply it an extra parameter. I mean the `$.fn[...]` part in the bottom. It is called when you execute `$('.search-select-ajax').aplusSearchSelectAjax({}, true)`.
I think it is best to understand the purpose of resetSelection and then fix it for the new ajax version even if we are not really using it now. I assume this would be quite easy to do anyway.

Reply:
This method is called only when editing submitters of an exercise. The edit submitters page has quick select for groups, and selecting a group will call the `resetSelection` method to empty the currently selected students.

The search field uses the old version of `search-select` widget. I noticed that many search fields wants to use only the students enrolled to the course, such as this, and it wouldn't make any sense that the submitters field would search students from all registered accounts. Perhaps we could add an option to search from courses students only using URL `/api/v2/courses/1/students/`. Or we can let those fields to use HTML templates with all the couse students, as they currently do. Opinions on this @Mankro ?

Reply:
So in short: should we use ajax-search in fields that need to use only the enrolled students?

Reply:
>  Perhaps we could add an option to search from courses students only using URL /api/v2/courses/1/students/

Yes indeed, we want to support different API endpoints. If a form should only search enrolled students in the course, then it uses the API endpoint for that. The JS code does not need to be changed since it reads the API URL from the HTML data attribute. The server can set the right API URL to the data attribute for each form.